### DISCUSSION TITLE
How to update database based on the rules

### DISCUSSION THREAD

**Discussion by anary** [microsoft/RulesEngine#389] [Q&A]
I have 2 questions about this rules engine.
If I want to check the database records with million records, what is the best practices to use this engine?
How can I update the database table on the rules action?

Thanks

**Comment by Zefek** [comment]
I think it would be better to use Specification pattern with some translation to SQL or LINQ. Load milion records from database will have performance impact and it is better to filter data in SQL server instead of load them to memory.
